### ISSUE TITLE
errors with "undefined reference to `__imp___gmpz_X' "

### ISSUE THREAD

**Issue by jimmybungalo** [m-ab-s/media-autobuild_suite#2352]
[logs.zip](https://github.com/m-ab-s/media-autobuild_suite/files/10312781/logs.zip)
i've tried running it a few times but i think it keeps on failing at this specific point, idk what the problem is. seeing a lot of ``undefined reference to `__imp___gmpz_X'``.

**Comment by Biswa96** [comment]
Try to install mingw-w64-gmp package.

**Comment by jimmybungalo** [comment]
literally the exact same thing happened

**Comment by jonathancolledge** [comment]
> Try to install mingw-w64-gmp package.

Sorry for being thick, but I have this error. You mean to install that in Windows proper? Not install in some sort of build environment?

**Comment by basilisk0315** [comment]
I have the same problem here.

My mabs install is completely up to date. I can see my gmp related files (eg gmp.h and libgmp.a) have a timestamp of last Monday 19:25:08. The issue is possibly something to do with this because I had a successful build the Saturday before.

How to revert the gmp package?

**Comment by Biswa96** [comment]
Reverting the gmp package may not solve the issue. It may even make things worse. In the latest update of gmp, shared library was set as default. See this commit for further details https://github.com/msys2/MINGW-packages/commit/f31bdf893a9a2c13fe924e58d6cb0d94891109d9. This issue only happens with static build.

I have not found a solution _yet_ but I can share the idea how this can be fixed probably. The core idea is that dllimport attribute has to be disabled somehow for static linking.

* By default, `__GMP_LIBGMP_DLL` is defined in $MINGW_PREFIX/include/gmp.h file. If I add `-U__GMP_LIBGMP_DLL` in CFLAGS emacs build fails, because it may enabled somewhere. 
* The `__GMP_LIBGMP_DLL` is not defined if `__GMP_WITHIN_CONFIGURE` is defined. But if I add `-D__GMP_WITHIN_CONFIGURE` in CFLAGS emacs build fails, because `GMP_NUMB_BITS` is not defined properly.

You can examine the gmp.h file for further examination. If you find any solution let us know.

**Comment by Biswa96** [comment]
Workaround: Change the line in $prefix/include/gmp.h like this (1 -> 0)

```diff
 /* Instantiated by configure. */
 #if ! defined (__GMP_WITHIN_CONFIGURE)
 #define _LONG_LONG_LIMB 1
-#define __GMP_LIBGMP_DLL  1
+#define __GMP_LIBGMP_DLL  0
 #endif
```

**Comment by Biswa96** [comment]
Workaround: Change the line in $prefix/include/gmp.h like this (1 -> 0)

```diff
 /* Instantiated by configure. */
 #if ! defined (__GMP_WITHIN_CONFIGURE)
 #define _LONG_LONG_LIMB 1
-#define __GMP_LIBGMP_DLL  1
+#define __GMP_LIBGMP_DLL  0
 #endif
```

**Only** try this if you are static linking.

**Comment by andrerfcsantos** [comment]
I'm also seeing this error.

- [logs.zip](https://github.com/m-ab-s/media-autobuild_suite/files/10327646/logs.zip)
- [build/librtmp-git/ab-suite.install.log](https://github.com/m-ab-s/media-autobuild_suite/files/10327654/ab-suite.install.log)

@Biswa96 Unfortunately that workaround didn't work for me. I did that change and ran `media-autobuild_suite.bat` again, should I do something else?

**Comment by jonathancolledge** [comment]
> I'm also seeing this error.
> 
> - [logs.zip](https://github.com/m-ab-s/media-autobuild_suite/files/10327646/logs.zip)
> 
> 
> @Biswa96 Unfortunately that workaround didn't work for me. I did that change and ran `media-autobuild_suite.bat` again, should I do something else?

Same for me ☹️

**Comment by ryankashi** [comment]
Hi! I'm also getting this exact same problem. Sorry I'm a total media-autobuild_suite noob here, just trying to get the av1_nvenc working with hardware acceleration for my setup!

**Comment by 1480c1** [comment]
> I erased all binaries to rebuild everything after a while...
> 
> [logs.zip](https://github.com/m-ab-s/media-autobuild_suite/files/10333711/logs.zip)
> 
> **build/librtmp-git/ab-suite.install.log**
> 
> ```
> ...
> ccache gcc -Wall -pipe -static-libgcc -static-libstdc++ -o rtmpdump.exe rtmpdump.o -Llibrtmp -lrtmp -LG:/MABS/local32/lib -lgnutls -lws2_32 -lws2_32 -lws2_32 -lgmp -latomic -ladvapi32 -lcrypt32 -lncrypt -lbcrypt -lcrypt32 -LG:/MABS/msys64/mingw32/lib -lnettle -lhogweed -lgmp -LG:/MABS/msys64/mingw32/lib -lnettle -ltasn1 -lz -lbrotlienc -LG:/MABS/msys64/mingw32/lib -lbrotlicommon -lbrotlidec -LG:/MABS/msys64/mingw32/lib -lbrotlicommon -lzstd  -lz -lws2_32 -lwinmm -lgdi32 
> G:/MABS/msys64/mingw32/bin/../lib/gcc/i686-w64-mingw32/12.2.0/../../../../i686-w64-mingw32/bin/ld.exe: librtmp/librtmp.a(rtmp.o):rtmp.c:(.text+0x4c2): undefined reference to `_imp____gmpz_init2'
> G:/MABS/msys64/mingw32/bin/../lib/gcc/i686-w64-mingw32/12.2.0/../../../../i686-w64-mingw32/bin/ld.exe: librtmp/librtmp.a(rtmp.o):rtmp.c:(.text+0x4db): undefined reference to `_imp____gmpz_set_ui'
> G:/MABS/msys64/mingw32/bin/../lib/gcc/i686-w64-mingw32/12.2.0/../../../../i686-w64-mingw32/bin/ld.exe: librtmp/librtmp.a(rtmp.o):rtmp.c:(.text+0x4e5): undefined reference to `_imp____gmpz_cmp'
> G:/MABS/msys64/mingw32/bin/../lib/gcc/i686-w64-mingw32/12.2.0/../../../../i686-w64-mingw32/bin/ld.exe: librtmp/librtmp.a(rtmp.o):rtmp.c:(.text+0x4ff): undefined reference to `_imp____gmpz_set'
> G:/MABS/msys64/mingw32/bin/../lib/gcc/i686-w64-mingw32/12.2.0/../../../../i686-w64-mingw32/bin/ld.exe: librtmp/librtmp.a(rtmp.o):rtmp.c:(.text+0x514): undefined reference to `_imp____gmpz_sub_ui'
> G:/MABS/msys64/mingw32/bin/../lib/gcc/i686-w64-mingw32/12.2.0/../../../../i686-w64-mingw32/bin/ld.exe: librtmp/librtmp.a(rtmp.o):rtmp.c:(.text+0x546): undefined reference to `_imp____gmpz_powm'
> G:/MABS/msys64/mingw32/bin/../lib/gcc/i686-w64-mingw32/12.2.0/../../../../i686-w64-mingw32/bin/ld.exe: librtmp/librtmp.a(rtmp.o):rtmp.c:(.text+0x557): undefined reference to `_imp____gmpz_cmp_ui'
> G:/MABS/msys64/mingw32/bin/../lib/gcc/i686-w64-mingw32/12.2.0/../../../../i686-w64-mingw32/bin/ld.exe: librtmp/librtmp.a(rtmp.o):rtmp.c:(.text+0x569): undefined reference to `_imp____gmpz_clear'
> G:/MABS/msys64/mingw32/bin/../lib/gcc/i686-w64-mingw32/12.2.0/../../../../i686-w64-mingw32/bin/ld.exe: librtmp/librtmp.a(rtmp.o):rtmp.c:(.text+0x643): undefined reference to `_imp____gmpz_init2'
> G:/MABS/msys64/mingw32/bin/../lib/gcc/i686-w64-mingw32/12.2.0/../../../../i686-w64-mingw32/bin/ld.exe: librtmp/librtmp.a(rtmp.o):rtmp.c:(.text+0x656): undefined reference to `_imp____gmpz_sizeinbase'
> G:/MABS/msys64/mingw32/bin/../lib/gcc/i686-w64-mingw32/12.2.0/../../../../i686-w64-mingw32/bin/ld.exe: librtmp/librtmp.a(rtmp.o):rtmp.c:(.text+0x6c8): undefined reference to `_imp____gmp_randinit_mt'
> G:/MABS/msys64/mingw32/bin/../lib/gcc/i686-w64-mingw32/12.2.0/../../../../i686-w64-mingw32/bin/ld.exe: librtmp/librtmp.a(rtmp.o):rtmp.c:(.text+0x6d9): undefined reference to `_imp____gmp_randseed_ui'
> G:/MABS/msys64/mingw32/bin/../lib/gcc/i686-w64-mingw32/12.2.0/../../../../i686-w64-mingw32/bin/ld.exe: librtmp/librtmp.a(rtmp.o):rtmp.c:(.text+0x6ed): undefined reference to `_imp____gmpz_urandomb'
> G:/MABS/msys64/mingw32/bin/../lib/gcc/i686-w64-mingw32/12.2.0/../../../../i686-w64-mingw32/bin/ld.exe: librtmp/librtmp.a(rtmp.o):rtmp.c:(.text+0x6f6): undefined reference to `_imp____gmp_randclear'
> G:/MABS/msys64/mingw32/bin/../lib/gcc/i686-w64-mingw32/12.2.0/../../../../i686-w64-mingw32/bin/ld.exe: librtmp/librtmp.a(rtmp.o):rtmp.c:(.text+0x74a): undefined reference to `_imp____gmpz_powm'
> G:/MABS/msys64/mingw32/bin/../lib/gcc/i686-w64-mingw32/12.2.0/../../../../i686-w64-mingw32/bin/ld.exe: librtmp/librtmp.a(rtmp.o):rtmp.c:(.text+0x77e): undefined reference to `_imp____gmpz_set_str'
> G:/MABS/msys64/mingw32/bin/../lib/gcc/i686-w64-mingw32/12.2.0/../../../../i686-w64-mingw32/bin/ld.exe: librtmp/librtmp.a(rtmp.o):rtmp.c:(.text+0x79b): undefined reference to `_imp____gmpz_clear'
> G:/MABS/msys64/mingw32/bin/../lib/gcc/i686-w64-mingw32/12.2.0/../../../../i686-w64-mingw32/bin/ld.exe: librtmp/librtmp.a(rtmp.o):rtmp.c:(.text+0x7ea): undefined reference to `_imp____gmpz_clear'
> G:/MABS/msys64/mingw32/bin/../lib/gcc/i686-w64-mingw32/12.2.0/../../../../i686-w64-mingw32/bin/ld.exe: librtmp/librtmp.a(rtmp.o):rtmp.c:(.text+0x848): undefined reference to `_imp____gmpz_clear'
> G:/MABS/msys64/mingw32/bin/../lib/gcc/i686-w64-mingw32/12.2.0/../../../../i686-w64-mingw32/bin/ld.exe: librtmp/librtmp.a(rtmp.o):rtmp.c:(.text+0xbaa): undefined reference to `_imp____gmpz_sizeinbase'
> G:/MABS/msys64/mingw32/bin/../lib/gcc/i686-w64-mingw32/12.2.0/../../../../i686-w64-mingw32/bin/ld.exe: librtmp/librtmp.a(rtmp.o):rtmp.c:(.text+0xc61): undefined reference to `_imp____gmpz_init2'
> G:/MABS/msys64/mingw32/bin/../lib/gcc/i686-w64-mingw32/12.2.0/../../../../i686-w64-mingw32/bin/ld.exe: librtmp/librtmp.a(rtmp.o):rtmp.c:(.text+0xcae): undefined reference to `_imp____gmpz_set_str'
> G:/MABS/msys64/mingw32/bin/../lib/gcc/i686-w64-mingw32/12.2.0/../../../../i686-w64-mingw32/bin/ld.exe: librtmp/librtmp.a(rtmp.o):rtmp.c:(.text+0xcc6): undefined reference to `_imp____gmpz_set_ui'
> G:/MABS/msys64/mingw32/bin/../lib/gcc/i686-w64-mingw32/12.2.0/../../../../i686-w64-mingw32/bin/ld.exe: librtmp/librtmp.a(rtmp.o):rtmp.c:(.text+0xce4): undefined reference to `_imp____gmpz_clear'
> G:/MABS/msys64/mingw32/bin/../lib/gcc/i686-w64-mingw32/12.2.0/../../../../i686-w64-mingw32/bin/ld.exe: librtmp/librtmp.a(rtmp.o):rtmp.c:(.text+0xd7d): undefined reference to `_imp____gmpz_init2'
> G:/MABS/msys64/mingw32/bin/../lib/gcc/i686-w64-mingw32/12.2.0/../../../../i686-w64-mingw32/bin/ld.exe: librtmp/librtmp.a(rtmp.o):rtmp.c:(.text+0xdd4): undefined reference to `_imp____gmpz_set_str'
> G:/MABS/msys64/mingw32/bin/../lib/gcc/i686-w64-mingw32/12.2.0/../../../../i686-w64-mingw32/bin/ld.exe: librtmp/librtmp.a(rtmp.o):rtmp.c:(.text+0xded): undefined reference to `_imp____gmpz_clear'
> G:/MABS/msys64/mingw32/bin/../lib/gcc/i686-w64-mingw32/12.2.0/../../../../i686-w64-mingw32/bin/ld.exe: librtmp/librtmp.a(rtmp.o):rtmp.c:(.text+0xe5b): undefined reference to `_imp____gmpz_sizeinbase'
> G:/MABS/msys64/mingw32/bin/../lib/gcc/i686-w64-mingw32/12.2.0/../../../../i686-w64-mingw32/bin/ld.exe: librtmp/librtmp.a(rtmp.o):rtmp.c:(.text+0xeba): undefined reference to `_imp____gmpz_powm'
> G:/MABS/msys64/mingw32/bin/../lib/gcc/i686-w64-mingw32/12.2.0/../../../../i686-w64-mingw32/bin/ld.exe: librtmp/librtmp.a(rtmp.o):rtmp.c:(.text+0xedb): undefined reference to `_imp____gmpz_clear'
> G:/MABS/msys64/mingw32/bin/../lib/gcc/i686-w64-mingw32/12.2.0/../../../../i686-w64-mingw32/bin/ld.exe: librtmp/librtmp.a(rtmp.o):rtmp.c:(.text+0x457b): undefined reference to `_imp____gmpz_clear'
> collect2.exe: error: ld returned 1 exit status
> make: *** [Makefile:78: rtmpdump] Error 1
> ```

From #2354

**Comment by 1480c1** [comment]
> Hi everyone!
> 
> I ran into this issue earlier today while compiling FFMPEG. Specifically, while compiling ffmpeg-git, I have received the error:
> 
> `ERROR: gmp not found`
> 
> Other than that, everything during the build process seemed to work fine. Of course, this is a pretty big blocker regarding compilations.
> 
> I am building 64-bit and have chosen the recommended build settings. In addition, I chose the "Light" build regarding the ffmpeg packages included. I have attached the .zip as requested as well.
> 
> What can be done to solve this error so that I can compile FFMPEG?
> 
> Thanks!
> 
> ```shell
> 01:16:08 Warning: We will not accept any issues lacking any form of logs or logs.zip!
> 01:16:08
>         Starting 64bit compilation of all tools
> Imported default FFmpeg options from .bat
> License: nonfree
> Imported default mpv options from .bat
> 01:16:48
>         Starting 64bit compilation of global tools
> 01:16:50
>         Starting 64bit compilation of audio tools
> 01:16:59   Running git update for fdk-aac...
> 01:16:59 fdk-aac git  ...................................... [Up-to-date]
> 01:17:03   Running git update for opus...
> 01:17:03 opus git  ......................................... [Up-to-date]
> 01:17:03
>         Starting 64bit compilation of video tools
> 01:17:09   Running git update for vpx...
> 01:17:09 vpx git  .......................................... [Up-to-date]
> 01:17:17   Running git update for aom...
> 01:17:18 aom git  .......................................... [Up-to-date]
> 01:17:23   Running git update for dav1d...
> 01:17:24 dav1d git  ........................................ [Up-to-date]
> 01:17:27   Running git update for SDL...
> 01:17:28 SDL git  .......................................... [Up-to-date]
> 01:17:32   Running git update for AMF...
> 01:17:32 AMF git  .......................................... [Up-to-date]
> 01:17:37   Running git update for x264...
> 01:17:38 x264 git  ......................................... [Up-to-date]
> 01:17:41   Running git update for x265_git...
> 01:17:42 x265_git git  ..................................... [Up-to-date]
> 01:17:46   Running git update for ffnvcodec...
> 01:17:46 ffnvcodec git  .................................... [Up-to-date]
> 01:17:48   Running git clone for ffmpeg...
> 01:19:16   Running git update for ffmpeg...
> 01:19:17 ┌ ffmpeg git  ............................... [Recently updated]
> 01:19:17 ├ Changing options to comply to nonfree...
> 01:19:20 ├ Compiling static FFmpeg...
> 01:19:22 ├ Running configure...
> Likely error (tail of the failed operation logfile):
> CXXFLAGS: -mthreads -mtune=generic -O2 -pipe
> LDFLAGS: -pipe -static-libgcc -static-libstdc++ -L/local64/lib -L/mingw64/lib
> ../configure --prefix=/local64 --bindir=/local64/bin-video --pkg-config=pkgconf --pkg-config-flags=--keep-system-libs --keep-system-cflags --static --cc=ccache gcc --cxx=ccache g++ --ld=ccache g++ --extra-cxxflags=-fpermissive --extra-cflags=-Wno-int-conversion --disable-autodetect --enable-amf --enable-bzlib --enable-cuda --enable-cuvid --enable-d3d11va --enable-dxva2 --enable-iconv --enable-lzma --enable-nvenc --enable-schannel --enable-zlib --enable-sdl2 --enable-ffnvcodec --enable-nvdec --enable-cuda-llvm --enable-gmp --enable-libmp3lame --enable-libopus --enable-libvorbis --enable-libvpx --enable-libx264 --enable-libx265 --enable-libdav1d --enable-libaom --disable-debug --enable-libfdk-aac --extra-libs=-liconv --enable-gpl --enable-version3 --enable-nonfree --extra-version=gf0d2ed135c+1
> ERROR: gmp not found
> 
> If you think configure made a mistake, make sure you are using the latest
> version from Git.  If the latest version fails, report the problem to the
> ffmpeg-user@ffmpeg.org mailing list or IRC #ffmpeg on irc.libera.chat.
> Include the log file "ffbuild/config.log" produced by configure as this will help
> solve the problem.
> configure failed. Check C:/ffmpeg_installer_new/build/ffmpeg-git/build-static-64bit/ab-suite.configure.log
> This is required for other packages, so this script will exit.
> 01:21:39   Creating diagnostics file...
> 
> Attach C:\ffmpeg_installer_new\build\logs.zip to the GitHub issue.
> Make sure the suite is up-to-date before reporting an issue. It might've been fixed already.
> 
> Try running the build again at a later time.
> ```
> 
> [logs.zip](https://github.com/m-ab-s/media-autobuild_suite/files/10335471/logs.zip)

from #2355

**Comment by RivenSkaye** [comment]
I've tried the suggested hacky workaround, as well as compiling GMP from source myself, but it seems like mingw is disallowing the use of undefined symbols in shared libraries (including headers) which is forcing DLL builds of libgmp to occur.
Perhaps we can look into compiling an older version of GMP for the time being, e.g. 6.1 or 6.0?

**Comment by Biswa96** [comment]
> Perhaps we can look into compiling an older version of GMP for the time being, e.g. 6.1 or 6.0?

The issue is not about any version of GMP. Recently, the mingw-w64-gmp package was built to favor shared library in this commit https://github.com/msys2/MINGW-packages/commit/f31bdf893a9a2c13fe924e58d6cb0d94891109d9. So, GMP has to be compiled with static libraries only.

**Comment by RivenSkaye** [comment]
I think you missed the part in my comment where the latest stable from their Mercurial repo (6.2) refuses to statically compile on mingw _at all_, which is why I suggested compiling a previous version to work around this

**Comment by ryankashi** [comment]
If I use the included .bat from an earlier version of media-autobuild_suite, will I be able to build ffmpeg without getting this error?

Otherwise, for the time being, how would I use this .bat in order to build ffmpeg?

**Comment by LigH-de** [comment]
The media-autobuild_suite.bat file is mainly used to configure the MinGW shell script build/media-suite_compile.sh which automates the building of all the components and finally ffmpeg. But the reason for issues most often lies within the components' sources, not in the files of the suite: Something is changed which breaks compatibility to ffmpeg or to other related components. Using an older batch file does not revert the changes in the downloaded source files of the compiled software.

**Comment by H7-26** [comment]
Switching back to gmp-6.2.1-3 solved all my gmp problems (for now).
If you want to try it, insert the following line into line 3 of the **build\media-suite_compile.sh** file:
for 64-bit:
`pacman -U --noconfirm https://repo.msys2.org/mingw/mingw64/mingw-w64-x86_64-gmp-6.2.1-3-any.pkg.tar.zst`
for 32-bit:
`pacman -U --noconfirm https://repo.msys2.org/mingw/mingw32/mingw-w64-i686-gmp-6.2.1-3-any.pkg.tar.zst`
I have only tested it for 64-bit Clang & GCC

**Comment by andrerfcsantos** [comment]
@H7-26 Thanks, thar fixed it for me.

**Comment by wbtcpip2** [comment]
Unfortunately not working for i686

**Comment by Ajaja** [comment]
@H7-26  Thanks, it works. 
You can also just add this package to ```IgnorePkg``` in ```pacman.conf``` after downgrading.

**Comment by RomaKoks** [comment]
@Ajaja could you please explain in more details how to do that?
I've tried to add in `msys64/etc/pacman.conf`
`IgnorePkg   = gmp` and `IgnorePkg   = mingw-w64-x86_64-gmp` 
but still get an error...
(row in `build\media-suite_compile.sh` file was inserted according to @H7-26 advise)

I will attach [logs.zip](https://github.com/m-ab-s/media-autobuild_suite/files/10376298/logs.zip) with hope that this can help somehow

**Comment by RomaKoks** [comment]
@Ajaja Yes, you are right, after I re-cloned the repository and performed the specified actions - I got to the problem with vulcan. But maybe they are somehow connected? Although I tried again now, but without compiling shared libraries, and static compiled successfully. So yes, apparently this is a separate issue related to shared libraries. Thanks!

**Comment by ronald8192** [comment]
> Switching back to gmp-6.2.1-3 solved all my gmp problems (for now). If you want to try it, insert the following line into line 3 of the **build\media-suite_compile.sh** file: for 64-bit: `pacman -U --noconfirm https://repo.msys2.org/mingw/mingw64/mingw-w64-x86_64-gmp-6.2.1-3-any.pkg.tar.zst` for 32-bit: `pacman -U --noconfirm https://repo.msys2.org/mingw/mingw32/mingw-w64-i686-gmp-6.2.1-3-any.pkg.tar.zst` I have only tested it for 64-bit Clang & GCC

Doesn't work for me, still getting the same error logs:
```
17:43:28   Running git update for librtmp...
17:43:28 ┌ librtmp git  .............................. [Recently updated]
17:43:28 ├ Running install...
Likely error (tail of the failed operation logfile):
D:/media-autobuild_suite-master/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/media-autobuild_suite-master/local64/lib/libgnutls.a(provable-prime.o):provable-prime:(.text+0xbb6): undefined reference to `__imp___gmpz_set'
D:/media-autobuild_suite-master/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/media-autobuild_suite-master/local64/lib/libgnutls.a(provable-prime.o):provable-prime:(.text+0xc8c): undefined reference to `__imp___gmpz_add_ui'
D:/media-autobuild_suite-master/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/media-autobuild_suite-master/local64/lib/libgnutls.a(provable-prime.o):provable-prime:(.text+0xcca): undefined reference to `__imp___gmpz_clear'
D:/media-autobuild_suite-master/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/media-autobuild_suite-master/local64/lib/libgnutls.a(provable-prime.o):provable-prime:(.text+0xcda): undefined reference to `__imp___gmpz_clear'
D:/media-autobuild_suite-master/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/media-autobuild_suite-master/local64/lib/libgnutls.a(provable-prime.o):provable-prime:(.text+0xcea): undefined reference to `__imp___gmpz_clear'
D:/media-autobuild_suite-master/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/media-autobuild_suite-master/local64/lib/libgnutls.a(provable-prime.o):provable-prime:(.text+0xcfa): undefined reference to `__imp___gmpz_clear'
D:/media-autobuild_suite-master/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/media-autobuild_suite-master/local64/lib/libgnutls.a(provable-prime.o):provable-prime:(.text+0xd0a): undefined reference to `__imp___gmpz_clear'
D:/media-autobuild_suite-master/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/media-autobuild_suite-master/local64/lib/libgnutls.a(provable-prime.o):provable-prime:(.text+0xd1a): more undefined references to `__imp___gmpz_clear' follow
collect2.exe: error: ld returned 1 exit status
make: *** [Makefile:78: rtmpdump] Error 1
install failed. Check D:/media-autobuild_suite-master/build/librtmp-git/ab-suite.install.log
This is required for other packages, so this script will exit.
17:43:32   Creating diagnostics file...

Attach D:\media-autobuild_suite-master\build\logs.zip to the GitHub issue.
Make sure the suite is up-to-date before reporting an issue. It might've been fixed already.

Try running the build again at a later time.
```

[ab-suite.install.log](https://github.com/m-ab-s/media-autobuild_suite/files/10396523/ab-suite.install.log)

**Comment by Slapbox** [comment]
Builds fail with `ERROR: gmp not found`

Trying @H7-26's solution currently.

**Comment by jacobandgeckos** [comment]
Almost exactly the same as output as @ronald8192 under librtmp from me.

[logs.zip](https://github.com/m-ab-s/media-autobuild_suite/files/10430190/logs.zip)
[ab-suite.install.log](https://github.com/m-ab-s/media-autobuild_suite/files/10430197/ab-suite.install.log)

**Comment by wbtcpip2** [comment]
any hope it will be fixed?

**Comment by 1480c1** [comment]
Waiting on https://github.com/msys2/MINGW-packages/pull/14954 to see which way the wind blows, in the worst case, we will need to start compiling our own gmp, but it may still be possible to use upstream's packages.

**Comment by Slapbox** [comment]
@H7-26's solution worked for me. @1480c1 what's the problem with that?

**Comment by 1480c1** [comment]
> besides that not really being the proper place for the download to occur

Well, that is one of the points, but the main issue is the fact that because it's a different version than what other packages might be compiled against, we could silently have issues and not realize them, but also because that can break at anytime.
I don't recall the exact expiration policy of the main repo, but essentially, it would only be a bandage as long as the url is still alive. Thus, I can't easily fully endorse adding those lines to the main repo. If it's on a user-by-user basis personally adding it to their own clone, it's fine, but I don't really want to commit it to the code.

**Comment by Mikipa** [comment]
```
pacman -R mingw-w64-x86_64-gmp
cd build&&curl "https://gmplib.org/download/gmp/gmp-6.2.1.tar.xz" -so -|tar -xJvf -
cd gmp-6.2.1
./configure --prefix=/mingw64
make
make check
make install
sed -Ei 's/Version: .+/Version: 1.1.1/g' /local64/lib/pkgconfig/gnutls.pc
rm librtmp-git -r
```
and 
>` IgnorePkg  = gmp mingw-w64-x86_64-gmp`
in pacman.conf


after this steps is ok 

> 01:43:59   Running git update for librtmp...
> 01:43:59 ┌ librtmp git  .............................. [Recently updated]
> 01:43:59 ├ Running install...
> 01:44:03 ├ Stripping...
> 01:44:03 └ librtmp-git  ....................................... [Updated]
> 01:44:06   Running git clone for vpx...

**Comment by ELGarulli** [comment]
Hi I also tried the suggestion from H7-27 but still get the following:

`  Running git update for librtmp...
┌ librtmp git  ........................................ [Recently updated]
├ Running install...
Likely error (tail of the failed operation logfile):
C:/media-autobuild_suite/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:/media-autobuild_suite/local64/lib/libgnutls.a(provable-prime.o):provable-prime:(.text+0xbb6): undefined reference to `__imp___gmpz_set'
C:/media-autobuild_suite/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:/media-autobuild_suite/local64/lib/libgnutls.a(provable-prime.o):provable-prime:(.text+0xc8c): undefined reference to `__imp___gmpz_add_ui'
C:/media-autobuild_suite/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:/media-autobuild_suite/local64/lib/libgnutls.a(provable-prime.o):provable-prime:(.text+0xcca): undefined reference to `__imp___gmpz_clear'
C:/media-autobuild_suite/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:/media-autobuild_suite/local64/lib/libgnutls.a(provable-prime.o):provable-prime:(.text+0xcda): undefined reference to `__imp___gmpz_clear'
C:/media-autobuild_suite/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:/media-autobuild_suite/local64/lib/libgnutls.a(provable-prime.o):provable-prime:(.text+0xcea): undefined reference to `__imp___gmpz_clear'
C:/media-autobuild_suite/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:/media-autobuild_suite/local64/lib/libgnutls.a(provable-prime.o):provable-prime:(.text+0xcfa): undefined reference to `__imp___gmpz_clear'
C:/media-autobuild_suite/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:/media-autobuild_suite/local64/lib/libgnutls.a(provable-prime.o):provable-prime:(.text+0xd0a): undefined reference to `__imp___gmpz_clear'
C:/media-autobuild_suite/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:/media-autobuild_suite/local64/lib/libgnutls.a(provable-prime.o):provable-prime:(.text+0xd1a): more undefined references to `__imp___gmpz_clear' follow
collect2.exe: error: ld returned 1 exit status
make: *** [Makefile:78: rtmpdump] Error 1
install failed. Check C:/media-autobuild_suite/build/librtmp-git/ab-suite.install.log
This is required for other packages, so this script will exit.
  Creating diagnostics file...

Attach C:\media-autobuild_suite\build\logs.zip to the GitHub issue.
Make sure the suite is up-to-date before reporting an issue. It might've been fixed already.

Try running the build again at a later time.
`

[logs.zip](https://github.com/m-ab-s/media-autobuild_suite/files/10518794/logs.zip)

**Comment by joaquinito2070** [comment]
[compile.log](https://github.com/m-ab-s/media-autobuild_suite/files/10529649/compile.log)
[logs.zip](https://github.com/m-ab-s/media-autobuild_suite/files/10529651/logs.zip)

**Comment by Leo-van-Stee** [comment]
Hi, great that this autobuild system exists! I want to make small change in reporting of VMAF values, but compiling ffmpeg looked impossible. It should be doable with you 'autobuild'.
However, I'm running into the same error with librtmp. I see there was an update to one of the files 4 days ago, but I still get the error. I have tried all solutions mentioned, but to no avail... Will there be a solution?

Greetings and thanks!

┌ librtmp git  .............................. [Recently updated]
├ Running install...
Likely error (tail of the failed operation logfile):
D:/media-autobuild_suite-master/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/media-autobuild_suite-master/local64/lib/libgnutls.a(provable-prime.o):provable-prime:(.text+0xbb6): undefined reference to `__imp___gmpz_set'
.....
make: *** [Makefile:78: rtmpdump] Error 1
install failed. Check D:/media-autobuild_suite-master/build/librtmp-git/ab-suite.install.log
This is required for other packages, so this script will exit.

**Comment by 1480c1** [comment]
Seems like I need to move the modification to the header up, unfortunately, I cannot do anything since my machines are off

**Comment by Leo-van-Stee** [comment]
Glad you're looking into it. I hope you find a solution... I'm already trying to decipher how ffmpeg/libvmaf is outputting its data.

What I'm trying to achieve is the following: build a GUI around ffmpeg that reads the vmaf data as it is 'produced live', instead of after the analysis. It works for the bitrate: I can make ffmpeg output the framesizes, timestamps, etc. line-by-line and read the output using TProcess and Bytesavailable as soon as data is 'outputted' in the 'console'. This gives me a live update (even for multiple files in multiple processes).

I want to do the same for VMAF.

Now I'm looking for the right spot in ffmpeg/libvmaf to print a line to the standard output with the VMAF score each time a frame is calculated... Still going through all the functions/procedures to find the right spot. I.e. this should be somewhere in the slowest loop that iterates through all the pictures.. Is it vmaf_read_pictures? Any help is appreciated.

Greetings!

**Comment by Leo-van-Stee** [comment]
> It is working now. I made a zeranoe build and librtmp is created correctly! 
Thanks and greetings.
